samcli/lib/package/utils.py

```diff
@@ -242,28 +255,47 @@ def make_zip(file_name, source_root):
                 for filename in files:
                     full_path = os.path.join(root, filename)
                     relative_path = os.path.relpath(full_path, source_root)
-                    if platform.system().lower() == "windows":
-                        with open(full_path, "rb") as data:
-                            file_bytes = data.read()
-                            info = zipfile.ZipInfo(relative_path)
+                    with open(full_path, "rb") as data:
+                        file_bytes = data.read()
+                        info = zipfile.ZipInfo(relative_path)
+                        # Context: Nov 2020
+                        # Set external attr with Unix 0755 permission
+                        # Originally set to 0005 in the discussion below
+                        # https://github.com/aws/aws-sam-cli/pull/2193#discussion_r513110608
+                        # Changed to 0755 due to a regression in https://github.com/aws/aws-sam-cli/issues/2344
+                        # Final PR: https://github.com/aws/aws-sam-cli/pull/2356/files
+                        if (file_permissions and dir_permissions) or platform.system().lower() == "windows":
                             # Clear external attr set for Windows
                             info.external_attr = 0
-                            # Set external attr with Unix 0755 permission
-                            # Originally set to 0005 in the discussion below
-                            # https://github.com/aws/aws-sam-cli/pull/2193#discussion_r513110608
-                            # Changed to 0755 due to a regression in https://github.com/aws/aws-sam-cli/issues/2344
-                            # Mimicking Unix permission bits and recommanded permission bits
-                            # in the Lambda Trouble Shooting Docs
-                            info.external_attr = 0o100755 << 16
                             # Set host OS to Unix
                             info.create_system = 3
+                            info.external_attr = (
+                                0o100755 << 16
+                                if platform.system().lower() == "windows"
```

Comment:
Do we still need this **if windows** statement, given that we are already calling this with `755` permissions for windows at line 294?